docs/workflows/common-workflow-steps/custom-script.md

```diff
@@ -227,87 +227,133 @@ sudo -E printenv
+EOF
+
+# Restrict permissions for the .msmtprc file to avoid security issues
+chmod 600 ~/.msmtprc
+
+# Export mail variables to the shell environment
+echo "export MAIL_SERVER=$HOST" >> ~/.$(basename $SHELL)rc
+echo "export MAIL_PORT=$PORT" >> ~/.$(basename $SHELL)rc
+echo "export MAIL_USE_TLS=True" >> ~/.$(basename $SHELL)rc
+echo "export MAIL_USE_SSL=False" >> ~/.$(basename $SHELL)rc
+echo "export MAIL_USERNAME=$EMAIL_" >> ~/.$(basename $SHELL)rc
+echo "export MAIL_PASSWORD=$PASSWORD_" >> ~/.$(basename $SHELL)rc
+
+echo "From: $EMAIL_FROM
+To: $EMAIL_TO
+Subject: $EMAIL_SUBJECT
+
+$EMAIL_BODY" | msmtp --debug --from=$EMAIL_ -t $EMAIL_TO
 
-echo "export MAIL_SERVER = smtp.gmail.com
-export MAIL_PORT = 587
-export MAIL_USE_TLS = True
-export MAIL_USE_SSL = False
-export MAIL_USERNAME = $EMAIL_
-export MAIL_PASSWORD = $PASSWORD_" >> ~/.$(basename $SHELL)rc
 
 ```
 
 :::caution Credentials
 
 When using your own SMTP server credentials for the three variables below, please use Environment Variables. This prevents sensitive information, such as passwords, from being exposed to unauthorized individuals. For more detailed information, please refer to the [**Environment Variables**](/environment-variables/managing-variables) documentation.
 
-- $EMAIL_
-- $USERNAME_
-- $PASSWORD_
+- **$EMAIL_**: SMTP Server email address
+- **$USERNAME_**: Your email address
+- **$PASSWORD_**: Your email address password
+
+Otherwise, to send an e-mail you need to have some information such as e-mail subject, sender e-mail, reciever e-mail. You can use these parameters to use 
+
+- **EMAIL_SUBJECT**: Subject of sending e-mail
+- **EMAIL_TO**: Reciever e-mail address.
+- **EMAIL_FROM**: Sender e-mail address.
+- **EMAIL_BODY**: Content of sending e-mail
+
+:::
+
+
+:::caution Protocols and SMTP Host
+
+This script uses the TLS protocol for SMTP server usage. Since **Gmail SMTP** server is used in the script, the required protocols are pulled from **Google's SMTP** server using the `$HOST_` parameter. 
+If you are using your own SMTP server. Don't forget to change the `$HOST_` value here. 
+
+On the other hand, to change **TLS or SSL** usage, you can change the protocol by setting the `MAIL_USE_TLS` or `MAIL_USE_SSL` parameters in the script to `true/false`. Note that you need to change the `$PORT_` parameter when using **SSL and TLS**. For more information about protocols, please visit the [**Google's TLS and SSL**](https://support.google.com/a/answer/100181) documentation.
+
+:::
+
+
+:::danger Custom Email
+
+To use an SMTP server, this script first installs the necessary certificates, then authenticates to the server with the required credentials and sends the prepared email content to the recipient's email address. In order to change the sender's email address, the SMTP server must allow it by providing the necessary permissions. Otherwise, SMTP servers will send the email using the authenticated email address to prevent spoofing (impersonating someone else).
```

Comment:
@ftmzhrsckldr Actually I think yes, because this script can be used by a person who does not have any idea to work with this script. So, I explain the whole necessary cases.